src/issue.rs

```diff
@@ -10,46 +10,37 @@ use std::collections::HashMap;
 const ISSUE_NEEDS_A_PROJECT_MESSAGE: &str =
 	"{1} needs to be attached to a project or it will be closed.";
 
-fn issue_actor_and_project(
+async fn issue_actor_and_project(
 	issue: &github::Issue,
 	github_bot: &GithubBot,
 ) -> Result<Option<(github::User, github::Project)>> {
 	let repo = &issue.repository;
 	let issue_number = issue.number.context(error::MissingData)?;
-	github_bot
+	let actor_and_card = github_bot
 		.issue_events(&repo.name, issue_number)
-		.map(|issue_events| {
-			issue_events
-				.iter()
-				.sorted_by_key(|ie| ie.created_at)
-				.rev()
-				.find(|issue_event| {
-					issue_event.event == Some(github::Event::AddedToProject)
-						|| issue_event.event
-							== Some(github::Event::RemovedFromProject)
-				})
-				.and_then(|issue_event| {
-					if issue_event.event == Some(github::Event::AddedToProject)
```

Comment:
If this is `RemovedFromProject`, there is no project currently attached.

Reply:
I understand that, however nothing happens when it is `RemovedFromProject`. It becomes `None` in the `else` branch anyway? Is there missing functionality for that `else` branch or am I missing something?

Reply:
If there is no project attached we return `None` and handle it outside the function.  We check for `Removed` because if we didn't then the function might return a project that wasn't actually attached.  So the point isn't to act on `Removed` in this function, it is to catch that particular case and return None.

Reply:
Sorry I'm still not understanding. Let's say an event is not `AddedToProject` or `RemovedFromProject` this function will return `Ok(None)`. If it is `RemovedFromProject` it will also be `Ok(None)` so how do you tell the difference?

Reply:
We don't need to tell the difference.

`find` will return the first (IE most recent) event that is `Added` or `Removed`, or if neither of those exists then `None`.  If the most recent event is `Removed`, then we subsequently return `None`.  Both those cases mean there is no project attached, so we don't need to distinguish them.

If we didn't check for `Removed` in the `find`, it might return an obsolete `Added`, ignoring a more recent `Removed`, and the function would return an unattached project, which we would incorrectly treat as if attached.

Reply:
Okay, now I understand. I'll add back that functionality. Thanks.

Reply:
Now that I think about it, we should handle removing a project similarly to attaching one, since both should require confirmation from the owner.  But that will require other changes, so yeah for this PR it should stay as is.